tests/adv

```diff
@@ -20,118 +20,115 @@
-    test "${k}" = "$(jose jwk thp -a "${THP_DEFAULT_HASH}" -i "${k}")".jwk
-    # Rotate the key.
-    mv -f -- "${k}" ".${k}"
-done
-cd -
-fetch /adv
-
-# Lets's now test with multiple pairs of keys.
-for i in 1 2 3 4 5 6 7 8 9; do
-    tangd-keygen "${TMP}"/db other-sig-${i} other-exc-${i}
-    # Make sure the requested keys exist and are valid.
-    validate_sig "${TMP}/db/other-sig-${i}.jwk"
-    validate_exc "${TMP}/db/other-exc-${i}.jwk"
-
-    # Make sure keys generated by tangd-keygen have proper permissions.
-    valid_key_perm "${TMP}/db/other-sig-${i}.jwk"
-    valid_key_perm "${TMP}/db/other-exc-${i}.jwk"
-done
-
-# Verify the advertisement is correct.
-validate "$(fetch /adv)"
-
-# And make sure we can fetch an adv by its thumbprint.
-for jwk in "${TMP}"/db/other-sig-*.jwk; do
-    for alg in $(jose alg -k hash); do
-        fetch /adv/"$(jose jwk thp -a "${alg}" -i "${jwk}")" | ver "${jwk}"
+    THP_DEFAULT_HASH=S256     # SHA-256.
+    test "$(tang-show-keys $PORT)" = "$(jose jwk thp -a "${THP_DEFAULT_HASH}" -i $TMP/db/sig.jwk)"
+
+    # Check that new keys will be created if none exist.
+    rm -rf "${TMP}/db" && mkdir -p "${TMP}/db"
+    fetch /adv
+
+    # Now let's make sure the new keys were named using our default thumbprint
+    # hash and then rotate them and check if we still create new keys.
+    cd "${TMP}/db"
+    for k in *.jwk; do
+        # Check for the key name (SHA-256).
+        test "${k}" = "$(jose jwk thp -a "${THP_DEFAULT_HASH}" -i "${k}")".jwk
+        # Rotate the key.
+        mv -f -- "${k}" ".${k}"
     done
-done
-
-# Now let's test keys rotation.
-tangd-rotate-keys -d "${TMP}/db"
-for i in 1 2 3 4 5 6 7 8 9; do
-    # Make sure keys were excluded from advertisement.
-    validate_sig "${TMP}/db/.other-sig-${i}.jwk"
-    validate_exc "${TMP}/db/.other-exc-${i}.jwk"
-done
-
-# And test also that we have valid keys after rotation.
-thp=
-for jwk in "${TMP}"/db/*.jwk; do
-    validate_sig "${jwk}" && thp="$(jose jwk thp -a "${THP_DEFAULT_HASH}" \
-                                    -i "${jwk}")"
-
-    # Make sure keys generated by tangd-rotate-keys have proper permissions.
-    valid_key_perm "${jwk}"
-done
-[ -z "${thp}" ] && die "There should be valid keys after rotation"
-test "$(tang-show-keys $PORT)" = "${thp}"
+    cd -
+    fetch /adv
+
+    # Lets's now test with multiple pairs of keys.
+    for i in 1 2 3 4 5 6 7 8 9; do
+        tangd-keygen "${TMP}"/db other-sig-${i} other-exc-${i}
+        # Make sure the requested keys exist and are valid.
+        validate_sig "${TMP}/db/other-sig-${i}.jwk"
+        validate_exc "${TMP}/db/other-exc-${i}.jwk"
+
+        # Make sure keys generated by tangd-keygen have proper permissions.
+        valid_key_perm "${TMP}/db/other-sig-${i}.jwk"
+        valid_key_perm "${TMP}/db/other-exc-${i}.jwk"
+    done
+
+    # Verify the advertisement is correct.
+    validate "$(fetch /adv)"
+
+	# And make sure we can fetch an adv by its thumbprint.
```

Comment:
Why is there extra identation in this block of code (lines 108-132) ?

Reply:
Fixed.